### PR TITLE
proc_manager: replace poll thread by workqueue plus timer 

### PR DESCRIPTION
**Pull request by uLipe**

### CODE REVIEW

**Comment by uLipe** [comment]
@microbuilder PTAL

File: Kconfig.step [hunk 1]
```diff
@@ -67,40 +67,28 @@ config STEP_PROC_MGR_NODE_LIMIT
...
-	int "Priority level for the polling thread."
-	default 0 if STEP_PROC_MGR_EVENT_DRIVEN
-	default 7
+	int "Priority level for the polling handler."
+	default 0
+	range -4 0
 	help
-	  Sets the priority level for the polling thread that is created when
-	  STEP_PROC_MGR_POLL_RATE is > 0. A positive value indicates that this is
+	  Sets the priority level for the polling handler.A positive value indicates that this is
```
Comment (microbuilder) [inline Kconfig.step:85]: Do we want to limit this to a range of -4 and 0?

File: include/step/proc_mgr.h [hunk 2]
```diff
@@ -143,6 +143,7 @@ int step_pm_process(struct step_measurement *mes, int *matches, bool free);
  *        process, and processes them on a first in, first processed basis.
  *
  * @param mcnt      Pointer to the number of samples read from the sample pool.
+ *                  (Deprecated), currrently noop.
```
Comment (microbuilder) [inline include/step/proc_mgr.h:146]: Typo

File: include/step/sample_pool.h [hunk 3]
```diff
@@ -103,6 +95,20 @@ struct step_measurement *step_sp_alloc(uint16_t sz);
  */
 int32_t step_sp_bytes_alloc(void);
 
+/**
+ * @brief Checks how many messages are pending in the sample FIFO.
+ *
+ * @return positive number indicating the amout of pendin messages, 0 otherwise.
```
Comment (microbuilder) [inline include/step/sample_pool.h:101]: Pending

**Review by microbuilder — CHANGES_REQUESTED** [review]

**Comment by microbuilder** [comment]
@uLipe Can you update the Zephyr header references to start with `zephyr/zephyr.h` (etc.)? With the latest Zephyr code, builds will fail:

```
/Users/kevintownsend/Linaro/zephyr/modules/lib/step/tests/src/floatcheck.h:17:10: fatal error: zephyr.h: No such file or directory
   17 | #include <zephyr.h>
      |          ^~~~~~~~~~
compilation terminated.
[16/139] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj
[17/139] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32c_sw.c.obj
[18/139] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj
[19/139] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj
[20/139] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj
[21/139] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj
ninja: build stopped: subcommand failed.

INFO    - Total complete:    1/   1  100%  skipped:    0, failed:    0
INFO    - 0 of 1 test configurations passed (0.00%), 1 failed, 0 skipped with 0 warnings in 4.66 seconds
INFO    - In total 21 test cases were executed, 0 skipped on 1 out of total 488 platforms (0.20%)
INFO    - 0 test configurations executed on platforms, 1 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /Users/kevintownsend/Linaro/zephyr/modules/lib/step/twister-out/twister.json
INFO    - Writing xunit report /Users/kevintownsend/Linaro/zephyr/modules/lib/step/twister-out/twister.xml...
INFO    - Writing xunit report /Users/kevintownsend/Linaro/zephyr/modules/lib/step/twister-out/twister_report.xml...
INFO    - Run completed
```

**Comment by uLipe** [comment]
Ahhh that is true I suffered same problem inside of other PR I submit.

Will update it.

File: Kconfig.step (hunk 1 above)
Comment (uLipe) [reply Kconfig.step:85]: I hope so, because the idea of nodes are to execute as soon the get deferred to avoid high congestion in the worqueue.

Comment (microbuilder) [reply Kconfig.step:85]: Perhaps remove the comment on positive values, then, if we’re setting a ceiling of 0, and just state the range?

**Comment by uLipe** [comment]
@microbuilder PTAL again.

**Review by microbuilder — CHANGES_REQUESTED** [review]
Looks like there are still some header errors here:

```bash
$ twister --inline-logs \
  -p qemu_cortex_m3 \
  -T tests
```

```bash
/Users/kevintownsend/Linaro/zephyr/modules/lib/step/tests/src/floatcheck.h:17:10: fatal error: zephyr.h: No such file or directory
   17 | #include <zephyr.h>
```

**Comment by microbuilder** [comment]
@uLipe I was going to make a PR for the Zephyr header cleanup since it's quick to do, but that will make a conflict with the partial changes you've started here, so lets just roll it all up into this PR. Thanks!

**Comment by microbuilder** [comment]
Also ... further changes likely required in the future, renaming `<zephyr/zephyr.h` to `<zephyr/kernel.h>` but we can sort that out when the changes is merged in and we've moved to a major release using the new name. 

See: https://github.com/zephyrproject-rtos/zephyr/pull/49510

**Comment by microbuilder** [comment]
@uLipe That PR above was just merged, so maybe we should switch to the latest Zephyr commit, and just update to `kernel.h` to solve this hopefully once and for all?

**Comment by uLipe** [comment]
Yes agreed, I will take the opportunity to update all zephyr headers, regardless samples or not (zephyr namespace is only required if you build twister, applications out of tree actually does not need it)

And then we can check the header cleanup.

**Comment by uLipe** [comment]
@microbuilder PTAL

**Review by microbuilder — APPROVED** [review]

**Pull request merged**